### DISCUSSION TITLE
Proposal : Logo for the project and new website design

### DISCUSSION THREAD

**Discussion by ghost** [SkriptLang/Skript#4920] [Suggestions] [answered]
![skript](https://user-images.githubusercontent.com/24407938/43416464-02cc9428-9441-11e8-92fe-4e1358736680.png)

I am willing to create a logo for this project. Please share opinions.

![web 1920 1](https://user-images.githubusercontent.com/24407938/43417604-190d5184-9444-11e8-98d9-232ff35a673a.png)
Also here's quick sketch for website design. I can develop the new design.

Let me know if there is no need for logo&/design.

**Comment by jaylawl** [comment]
I propose calling it `benSKu`. In all seriousness.

A better web-presence sure wouldn't hurt. Just yesterday i was attempting to help out somebody on the skunity forums. It turned out he was still using Skript 2.1.2, released in 2014 on bukkit - which he considered the `latest` version. He plain didn't know about this place.

Would love to help out on documentation aswell

**Comment by Nicofisi** [comment]
Damn man, the docs redesign looks amazing! I love the color choice, font, padding, everything.

A little bit of criticism though: I don't think the fork name is needed on the logo anymore, this fork is the only one actively being developed, and even the link was recently updated not to contain `bensku`.

~~[By the way, Skript theoretically has a logo already!](https://media.forgecdn.net/avatars/71/627/636163282994452816.png) Jk though, it's way too much 2010.~~

Please do continue and feel free to hit me up somewhere if you want to & need help with generating the actual docs.

**Comment by ghost** [comment]
I continued little bit and this is what it looks like now, going to continue later.

## Logo
![skript](https://user-images.githubusercontent.com/24407938/43425610-d4c6475c-945b-11e8-809d-29136b617808.png)

## Web design
![web 1920 2](https://user-images.githubusercontent.com/24407938/43425615-d72c49c4-945b-11e8-9945-412622855951.png)

**Comment by abandonedaccount6235** [comment]
when done, can you please pick a license for the designs? Preferably one compatible with skript's

**Comment by ghost** [comment]
Skript's GNU General Public License v3.0 license is fine for me. Should we create another repo under SkriptLang for design & developement process and with issue tracker to propose ideas or something?

Feel free to set new logo as organization avatar.

**Comment by ghost** [comment]
![skript documentation](https://user-images.githubusercontent.com/24407938/43457045-4aa73860-94ce-11e8-8761-2f007dfefce1.png)
![skript documentation frontpage](https://user-images.githubusercontent.com/24407938/43457046-4ad15244-94ce-11e8-8d66-14e05b6c644d.png)

**Comment by bensku** [comment]
New website design *and* new logo? Count me in!

It should be easy enough to plug in new CSS and HTML to docs generator. I suppose other improvements could be made to the generation itself, too.

@0x4d48 Are you familiar with Git? Could set up a repository for website templates if you want.

**Comment by ghost** [comment]
Yes, I am familiar with Git. Repo would be great.

**Comment by bensku** [comment]
Alright. Skript's current docs templates (https://github.com/SkriptLang/Skript/tree/master/docs) should have some ideas about how to use them for new styling. In case something is unclear, I can also work on integrating your work to them.

**Comment by ghost** [comment]
Is it possible to test `docs/` on local machine?

**Comment by Nicofisi** [comment]
Do you have a Discord account? I think it would be easier to discuss this if you join here https://discord.gg/V4qFVnh

**Comment by bensku** [comment]
https://github.com/SkriptLang/site-templates - I've dropped you invite that gives push access there.

It is possible to test the templates locally. Place the template folder (or create a symbolic link) named as doc-templates under plugins/Skript and run /sk gen-docs from server console.

Edit: Too many new messages, got confused :)

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/24407938/43462287-61a7bf20-94de-11e8-902d-96afe86f2f23.png)
@bensku Could you try maybe adding README.md file as first file in repo?

**Comment by abandonedaccount6235** [comment]
@0x4d48 i'm guessing you are handling the HTML/css, then? If so, keep in mind #1402 will need support for a section documentation page, too. If not, I could give it a shot -- I have mild webdev experience and am the author of #1402 anyway.

edit: ignore me i just saw this: `can develop the new design`

**Comment by abandonedaccount6235** [comment]
just dropping this here:
![image](https://user-images.githubusercontent.com/28607612/43481206-f0079da8-94ca-11e8-941e-76d7835b0b82.png)

**Comment by ghost** [comment]
Well, if you want to help me with this project, that's ok for me. I would have more resources to put in other projects :trollface:. 

Also if you have ideas regarding how website works (generating html, reading json), they're welcome. I wouldn't want though to do anything else than a new template.

**Comment by Nicofisi** [comment]
@Pikachu920 @xXAndrew28Xx did you mean just generating the docs to json (or using something like TuSKe docs) and then reading them via JS? If yes, on one hand it sounds better for the development purposes for sure, but on the other, kinda RIP non-javascript users, I'm pretty sure all the main languages generate docs that are just static html

**Comment by 28andrew** [comment]
Well for sure converting to json. It doesn't really matter to me whether the server parses the json and serves  the web content or the client parses it and changes the DOM.

**Comment by abandonedaccount6235** [comment]
in skript-dev channel it was decided to stay static

**Comment by supertassu** [comment]
> in skript-dev channel it was decided to stay static

Could you (or somebody else) justify this? For me (web professional) a react site loading and parsing JSON seems much simpler.

**Comment by bensku** [comment]
I wouldn't say it is simpler, considering Skript already has a HTML template engine for generating docs. Same template engine generates current JSON docs, too.

Aside that, using Javascript for static sites has a number of downsides. First, the Javascript and JSON data files will need to be fetched. That takes time, unless HTTP/2 PUSH is available. In addition to that, it requires CPU time from client's computer. Parsing, JIT-compiling and executing JS to construct a static site is, seriously, waste of client's processing power (and battery life).

For these reasons, *no* static site should use Javascript for basic page content. Even if it is more convenient, *please* think of your users. Unless the site is for internal use in, say, a company, it is **bad** idea to assume that users have CPU power to waste. That is to say nothing of users who disable Javascript, or do not receive it due to a network errors.

It is possible to write *fast* Javascript code. However, can it be faster than having no executable code in your website? I don't think so.

**Comment by AyhamAl-Ali** [comment, marked as answer]
The docs theme has changed already see #3935 and #4319, this theme is no longer needed.
